libraries/ESPSoftRx/ESPSoftRx.cpp

```diff
@@ -0,0 +1,93 @@
+/* 
+  ESPSoftRx - Timer1 based Software Serial RX for ESP8266
+
+  Copyright (c) 2015 Hristo Gochkov. All rights reserved.
+  This file is part of the esp8266 core for Arduino environment.
+ 
+  This library is free software; you can redistribute it and/or
+  modify it under the terms of the GNU Lesser General Public
+  License as published by the Free Software Foundation; either
+  version 2.1 of the License, or (at your option) any later version.
+
+  This library is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+  Lesser General Public License for more details.
+
+  You should have received a copy of the GNU Lesser General Public
+  License along with this library; if not, write to the Free Software
+  Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+*/
+#include "Arduino.h"
+#include "ESPSoftRx.h"
+#include <cbuf.h>
+
+extern "C" uint32_t interrupt_reg;
```

Comment:
Since you have put interrupt_reg into Arduino.h, no need to re-declare it here. Same for the other class.